File: src/Model/GeneralStatement/PrintStatement.java
package Model.GeneralStatement;

import Model.Expression.IExpression;
import Model.ProgramState;
import Model.Utils.IDictionary;
import Model.Utils.IList;

public class PrintStatement implements IStatement {

    private IExpression expression;

    public PrintStatement(IExpression expression) {
        this.expression = expression;
    }

    @Override
    public ProgramState execute(ProgramState state) {
        IList<Integer> out = state.getOut();
        IDictionary<String, Integer> symTable = state.getSymTable();
        int value = expression.eval(symTable, state.getHeap());
        out.add(value);
        return state;
    }

    public String toString() {
        return "print(" + expression + ");";
    }
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Statement/GeneralStatement/PrintStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language_CS_Interpreter.Model.Expression;

namespace Toy_Language_CS_Interpreter.Model.Statement.GeneralStatement
{
    class PrintStatement : IStatement
    {
        private IExpression expression;

        public PrintStatement(IExpression expression)
        {
            this.expression = expression;
        }

        public ProgramState Execute(ProgramState state)
        {
            int result = expression.Evaluate(state.SymbolTable);
            state.Output.Add(result);
            return state;
        }

        public override string ToString()
        {
            return "Print(" + expression + ")";
        }
    }
}
